fix and enhance pipeline_webserver.md (#36992)
huggingface/transformers: docs/source/en/pipeline_webserver.md

    return JSONResponse(output)

async def server_loop(q):

>     p = pipeline(task="fill-mask",model="google-bert/bert-base-uncased")

Reviewer: Pipe maybe ?

Author: sure

```bash
curl -X POST -d "Paris is the [MASK] of France." http://localhost:8000/
```

> The output looks like below:

Reviewer 2: ```suggestion
This should return the output below.
```

1. There is no batch size limit.
2. The timeout is reset on every queue fetch, so you could end up waiting much longer than the `timeout` value before processing a request. This would also delay the first inference request by that amount of time. The web server always waits 1ms even if the queue is empty, which is inefficient, because that time can be used to start inference. It could make sense though if batching is essential to your use case.

>     It would be better to have a single 1ms deadline, instead of resetting it on every fetch, we can enhance `server_loop` as below.

Reviewer 2: ```suggestion
    It would be better to have a single 1ms deadline, instead of resetting it on every fetch, as shown below.
```
